addons/l10n_ar_withholding/wizards/account_payment_register.py

```diff
@@ -0,0 +1,199 @@
+        payment_vals = super()._create_payment_vals_from_wizard(batch_result)
+        payment_vals['amount'] = self.net_amount
+        conversion_rate = self._get_conversion_rate()
+        sign = 1
+        if self.partner_type == 'supplier':
+            sign = -1
+        for line in self.withholding_ids:
+            if not line.name or line.name == '/':
+                if line.tax_id.l10n_ar_withholding_sequence_id:
+                    line.name = line.tax_id.l10n_ar_withholding_sequence_id.next_by_id()
+                else:
+                    raise UserError(_('Please enter withholding number for tax %s') % line.tax_id.name)
+            dummy, account_id, tax_repartition_line_id = line._tax_compute_all_helper()
+            balance = self.company_currency_id.round(line.amount * conversion_rate)
+            payment_vals['write_off_line_vals'].append({
+                    **self._get_withholding_move_line_default_values(),
+                    'name': line.name,
+                    'account_id': account_id,
+                    'amount_currency': sign * line.amount,
+                    'balance': sign * balance,
+                    'tax_base_amount': sign * line.base_amount,
+                    'tax_repartition_line_id': tax_repartition_line_id,
+            })
+
+        for base_amount in list(set(self.withholding_ids.mapped('base_amount'))):
+            withholding_lines = self.withholding_ids.filtered(lambda x: x.base_amount == base_amount)
+            nice_base_label = ','.join(withholding_lines.mapped('name'))
+            account_id = self.company_id.l10n_ar_tax_base_account_id.id
+            base_amount = sign * base_amount
+            cc_base_amount = self.company_currency_id.round(base_amount * conversion_rate)
+            payment_vals['write_off_line_vals'].append({
+                **self._get_withholding_move_line_default_values(),
+                'name': _('Base Ret: ') + nice_base_label,
+                'tax_ids': [Command.set(withholding_lines.mapped('tax_id').ids)],
+                'account_id': account_id,
+                'balance': cc_base_amount,
+                'amount_currency': base_amount,
+            })
+            payment_vals['write_off_line_vals'].append({
+                **self._get_withholding_move_line_default_values(),  # Counterpart 0 operation
+                'name': _('Base Ret Cont: ') + nice_base_label,
+                'account_id': account_id,
+                'balance': -cc_base_amount,
+                'amount_currency': -base_amount,
+            })
+
+        return payment_vals
+
+    def _get_conversion_rate(self):
+        self.ensure_one()
+        if  self.currency_id !=  self.source_currency_id:
+            return self.env['res.currency']._get_conversion_rate(
+                self.currency_id,
+                self.source_currency_id,
+                self.company_id,
+                self.payment_date,
```

Comment:
```suggestion
                self.date,
```
In Odoo, the rate will be the accounting date and not the payment date.

Reply:
The wiz has no date field only payment_date.